go/vt/vtctl/grpcvtctldserver/query.go

```diff
@@ -0,0 +1,300 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package grpcvtctldserver
+
+import (
+	"fmt"
+	"time"
+
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/protoutil"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+	"vitess.io/vitess/go/vt/vtctl/schematools"
+
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtctldatapb "vitess.io/vitess/go/vt/proto/vtctldata"
+	"vitess.io/vitess/go/vt/proto/vttime"
+)
+
+// 2023-08-05 07:15:16
+const mysqlTimestampLayout = "2006-01-02 15:04:05"
+
+// TODO: explicit column list to guarantee parsing stability
+func selectSchemaMigrationsQuery(condition, order, skipLimit string) string {
+	return fmt.Sprintf(`select
+		*
+		from _vt.schema_migrations where %s %s %s`, condition, order, skipLimit)
+}
+
+func rowToSchemaMigration(row sqltypes.Row) (sm *vtctldatapb.SchemaMigration, err error) {
+	row = row[1:] // We're off by 1 because of the extra Tablet field added to the grouped query result
```

Comment:
we're also not currently doing anything with that extra `Tablet` field, which is a net-loss from the old output (i'm not necessarily saying that's a _bad_ thing, just noting it so everyone is aware of this. we can put it back, or leave it off, or mystery 3rd option)

Reply:
Not sure if we should use the `Tablet` field, as it just happens to be the `PRIMARY` of the shard, but does not otherwise have any specific meaning to the results.